lib/eco/autolboxdetector.py

```diff
@@ -82,6 +82,31 @@ def detect_lbox(self, errornode):
             errornode.autobox = None
             return False
 
+    def line_search(self, errornode):
+        valid = []
+        pv = self.op.prev_version
+        for sub in self.langs:
+            lbox = MagicTerminal("<{}>".format(sub))
+            node = errornode.prev_term
+            while True:
+                element = self.op.syntaxtable.lookup(node.state, lbox)
+                if type(element) in [Reduce, Shift]:
+                    r = self.langs[sub]
+                    r.mode_limit_tokens_new = self.mode_limit_tokens_new
+                    start = node.next_term
+                    result = r.parse(start)
+                    if r.possible_ends:
+                        for e in r.possible_ends:
+                            if e.lookup == "<ws>" or e.lookup == "<return>":
+                                continue
+                            if (self.contains_errornode(start, e, errornode) \
+                                    and self.parse_after_lbox_h2(lbox, e, start, pv)):
```

Comment:
This formatting is very hard to read. Maybe too much indentation or ...?

Reply:
I'm not sure how I can improve this.

Reply:
A nested `if` would probably be less worse. I had to read this 3 or 4 times to work out what was going on.